core/src/epicli/data/common/ansible/playbooks/roles/kubernetes_master/templates/canal.yml.j2

```diff
@@ -3454,6 +3454,19 @@ spec:
       labels:
         k8s-app: canal
     spec:
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+              - matchExpressions:
+                  - key: kubernetes.io/os
+                    operator: In
+                    values:
+                      - linux
+                  - key: kubernetes.io/arch
+                    operator: In
+                    values:
+                      - amd64
```

Comment:
Do we need that together with "nodeSelector" below?

Reply:
Yes as stated in the privious comment, one is for amd64, one is for aarch64

Reply:
I meant these lines just after node affinity which specifies the same:

```yaml
nodeSelector:
  kubernetes.io/os: linux
```

Reply:
Yeah we should be able to remove them.